mixins/jahia.yml

```
@@ -854,6 +858,133 @@ actions:
           fi
         user: tomcat
 
+  setupPat:
+    # Parameters:
+    # jahia_version: the jahia version
+    - generatePatAndKey
+    - savePatInVault:
+        __secret__pat_token: ${globals.__secret__pat_token}
+        __secret__pat_key: ${globals.__secret__pat_key}
+    - setPatInJahia:
+        __secret__pat_token: ${globals.__secret__pat_token}
+        jahia_version: ${this.jahiaVersion}
+    - setPatInHaproxy:
+        __secret__pat_token: ${globals.__secret__pat_token}
+
+  generatePatAndKey:
+    - cmd[proc]: |-
+        graph_token=$(cat /dev/urandom | tr -dc '[[:graph:]]' | fold -w 32 | head -n 1)
+        token=$(echo -n "$graph_token" | base64)
+        key=$(echo -n "$graph_token" | awk '{printf substr($1,0,16)}' | od -A n -t x1 | awk '{printf "%s%s%s%s-%s%s-%s%s-%s%s-%s%s%s%s%s%s",$1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16}')
+        printf '{"hideThisOutput": true, "token": "%s", "key": "%s"}' "$token" "$key"
+
+    - script: |-
+        __secret__response = ${response.out};
+        token = __secret__response["token"];
+        key = __secret__response["key"];
+
+        return {
+          "result": 0,
+          "onAfterReturn": {
+            setGlobals: {
+              "__secret__pat_token": token,
+              "__secret__pat_key": key,
+            }
+          }
+        };
+
+  savePatInVault:
+    # Parameters:
+    #  - __secret__pat_token: PAT token
+    #  - __secret__pat_key: PAT key
+    - getVaultData
+    - vaultSecretSet:
+        secretPath: "paas/customers/${globals.organizationName}/paas_${env.shortdomain}/PAT/root"
+        __secret__secretData: '{"token":"${this.__secret__pat_token}","key":"${this.__secret__pat_key}"}'
+
+  setPatInJahia:
+    # Parameters:
+    # __secret__pat_token: the personal access token
+    # jahia_version: the jahia version
+    - cmd[proc]: |-
+        __secret__pat_token=${this.__secret__pat_token}
+        groovy_file_path=/data/digital-factory-data/patches/groovy/pat.groovy
+
+        jahia_major_version=$(echo "${this.jahia_version}" | head -c1)
```

looks dangerous (although very unlikely), but I will prefer something like this:
```suggestion
        jahia_major_version=$(echo "${this.jahia_version}" | cut -d'.' -f1)
```

---

done